waiter/integration/waiter/token_request_test.clj

```diff
@@ -232,42 +275,96 @@
                     (str headers))
                 (delete-service waiter-url service-id))))
           (finally
-            (delete-token-and-assert waiter-url token)))))))
+            (delete-token-and-assert waiter-url token :query-params {"erase" true})))))))
+
+(deftest ^:parallel ^:integration-fast test-token-sync-invalid-run-as-user
+  (testing-using-waiter-url
+    (let [service-id-prefix (rand-name)]
+      (testing "token-syncing"
+        (testing "active-token"
+          (let [token (create-token-name waiter-url service-id-prefix)]
+            (try
+              (log/info "creating configuration using token" token)
+              (let [token-description {:health-check-url "/probe"
+                                       :name service-id-prefix
+                                       :owner (retrieve-username)
+                                       :run-as-user "foo-bar"
+                                       :token token
+                                       :update-mode "sync"}
+                    response (post-token waiter-url token-description)]
+                (assert-response-status response 200))
+              (log/info "created configuration using token" token)
+              (let [token-response (get-token waiter-url token)
+                    response-body (json/read-str (:body token-response))]
+                (is (contains? response-body "last-update-time"))
+                (is (= {"deleted" false, "health-check-url" "/probe", "name" service-id-prefix, "owner" (retrieve-username),
+                        "run-as-user" "foo-bar", "update-mode" "sync"}
+                       (dissoc response-body "last-update-time"))))
+              (log/info "asserted retrieval of configuration for token" token)
+              (finally
+                (delete-token-and-assert waiter-url token :query-params {"erase" true})))))
+
+        (testing "deleted-token"
+          (let [token (create-token-name waiter-url service-id-prefix)]
+            (try
+              (log/info "creating configuration using token" token)
+              (let [token-description {:deleted true
+                                       :health-check-url "/probe"
+                                       :name service-id-prefix
+                                       :owner (retrieve-username)
+                                       :run-as-user "foo-bar"
+                                       :token token
+                                       :update-mode "sync"}
```

Comment:
Thoughts on having `update-mode` be a query string parameter instead of adding more to the post body?